server/test/unit/auth/test_oauth.py

```diff
@@ -62,21 +62,34 @@ def jwks():
     data = dict(alg="RS256", kty="RSA", use="sig", kid="fake_kid",)
     return make_response(jsonify(dict(keys=[data])))
 
-# The port that the mock oauth server will listen on
-PORT = random.randint(10000, 12000)
-
 
 # function to launch the mock oauth server
-def launch_mock_oauth():
-    mock_oauth_app.run(port=PORT)
+def launch_mock_oauth(mock_port):
+    mock_oauth_app.run(port=mock_port)
 
 
 class AuthTest(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
+        # The port that the mock oauth server will listen on
+        cls.mock_port = random.randint(10000, 12000)
         cls.dataset_dataroot = FIXTURES_ROOT
-        cls.mock_oauth_process = Process(target=launch_mock_oauth)
+        cls.mock_oauth_process = Process(target=launch_mock_oauth, args=(cls.mock_port,))
         cls.mock_oauth_process.start()
+        server_okay = False
+        for _ in range(5):
```

Comment:
Can you add a comment explaining why we need to retry in order for this setup to work?

Reply:
Oops -- sorry I guess my comment wasn't clear. I was more wondering what necessitates the `for _ in range(5):` line (i.e. why do we need to retry). Does it take up to 5 seconds for the mock oauth server to be ready?

Reply:
All this is doing is polling until the mock server is ready.  The issue is we are starting a mock oauth server, then we are starting a cellxgene server, which will start making requests to the mock oauth server.   So there is a race condition because the mock oauth server needs to be ready before it gets requests.  So, we check to see if it is ready, and if not we wait 1 second, then try again.  If it gets to 5 seconds, which is shouldn't,  we assume something has gone wrong and fail the test.

Reply:
Ah got it -- this makes sense to me. Mind writing the same explanation as a comment above the for loop? That way, the next reader knows why we need that retry there.